public/app/core/utils/richHistory.ts

```diff
@@ -304,3 +312,46 @@ export function createDatasourcesList(queriesDatasources: string[]) {
   });
   return datasources;
 }
+
+export function notEmptyQuery(query: DataQuery) {
+  /* Check if query has any other properties besides key, refId and datasource.
+   * If not, then we consider it empty query.
+   */
+  const strippedQuery = _.omit(query, ['key', 'refId', 'datasource']);
+  const queryKeys = Object.keys(strippedQuery);
+
+  if (queryKeys.length > 0) {
+    return true;
+  }
+
+  return false;
+}
+
+/* These functions are created to migrate string queries (from 6.7 release) to DataQueries. They can be removed after 7.1 release. */
+function createDataQueriesRichHistory(richHistory: RichHistoryQuery[]) {
```

Comment:
I would probably call this `migrateRichHistory` or something in that lines to be clear what is going on.

Reply:
updated in https://github.com/grafana/grafana/pull/23627/commits/597b0a497c8a475ae9e872ea397f7dd8314bfdef